src/generic/hooks.jsx

```diff
@@ -0,0 +1,21 @@
+import { useSelector } from 'react-redux';
+import { getUserPermissions, getUserPermissionsEnabled } from './data/selectors';
+
+const useUserPermissions = () => {
+  const userPermissionsEnabled = useSelector(getUserPermissionsEnabled);
+  const userPermissions = useSelector(getUserPermissions);
+
+  const hasPermissions = (checkPermissions) => {
```

Comment:
consider renaming `checkPermissions` to improve readability of this piece of code.